scipy/optimize/linprog_ip.py

```diff
@@ -0,0 +1,1994 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Thu Feb 02 17:09:17 2017
+
+@author: Matt
+
+To do:
+multiple presolve
+questions for scipy-dev
+* This is written in Python 2. How should I check compatibility with Python 3?
+* Are the warnings OK? Do I need an option to disable them?
+* Is 100% pep8 compliance necessary?
+* How should unit tests be performed on both simplex and ip?
+* Can I leave multi-pass presolve to an update?
+* Can I leave sparse presolve to an update?
+* Do I need to implement a callback interface?
+* There are some changes to be made to simplex; it should use the ip presolve
+ routine. Can that be left to an update?
+* How do I document options not common with simplex?
```

Comment:
Regarding the name: the question is if there is only one algorithm known as "interior-point", or it's a generic name for a class of algorithms with different implementations. If it's the latter, *and* we might want to add more algorithms from this class later on, then we should not call this one "interior-point". Otherwise, it's not a problem.